src/SFA.DAS.EmployerFinance/Commands/RefreshEmployerLevyData/ILevyImportCleanupStrategy.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerFinance.Models.Levy;
+
+namespace SFA.DAS.EmployerFinance.Commands.RefreshEmployerLevyData
+{
+    /// <summary>
+    ///     Represents a service that can take levy declarations and pre-process them to make them 
+    ///     suitable for processing by EAS. 
+    /// </summary>
+    /// <remarks>
+    ///     The pre-processing that occurs is to remove duplicate subsidy ids, remove declarations that 
+    ///     are outside the period range and evaluate year end adjustments. 
+    /// </remarks>
+    public interface ILevyImportCleanerStrategy
+    {
+        Task<IEnumerable<DasDeclaration>> Cleanup(string empRef, IEnumerable<DasDeclaration> declarations);
```

Comment:
As the interface is really internal I'm not convinced we need to be so open on the input and output types given its forcing us to do a ToArray on both sides.
I'm not actually convinced we need an interface but that's a bigger discussion.

Reply:
Unfortunately the interface is injected into a public constructor (so must be public itself) and if it's public it should play nicely. 

I'm not sure that changing the input parameter to an array would not remove the need to call ToArray(); the inbound declarations are ordered by submission date (creating another enumeration) which would _still_ need to be projected to an array.

I've changed the return type to an array.